packages/js/components/src/date-time-picker-control/test/index.tsx

```diff
@@ -259,6 +261,99 @@ describe( 'DateTimePickerControl', () => {
 		);
 	}, 10000 );
 
+	// We need to bump up the timeout for this test because:
+	//     1. userEvent.type() is slow (see https://github.com/testing-library/user-event/issues/577)
+	//     2. moment.js is slow
+	// Otherwise, the following error can occur on slow machines (such as our CI), because Jest times out and starts
+	// tearing down the component while test microtasks are still being executed
+	// (see https://github.com/facebook/jest/issues/12670)
+	//       TypeError: Cannot read properties of null (reading 'createEvent')
+	it( 'should call the current onChange when the input is changed', async () => {
+		const originalDateTime = moment( '2022-09-15 02:30:40' );
+		const dateTimeFormat = 'HH:mm, MM-DD-YYYY';
+		const newDateTimeInputString = '02:04, 06-08-2010';
+		const newDateTime = moment( newDateTimeInputString, dateTimeFormat );
+		const originalOnChangeHandler = jest.fn();
+		const newOnChangeHandler = jest.fn();
+
+		let count = 0;
+
+		const Container: React.FC< { children?: React.ReactNode } > = ( {
+			children,
+		} ) => {
+			function getChildren() {
+				if ( typeof children === 'function' ) {
+					return children( {
+						// change the onChange handler after the initial render
+						onChange:
+							count++ === 0
+								? originalOnChangeHandler
+								: newOnChangeHandler,
+						className: 'foo',
+					} );
+				}
+
+				return children;
+			}
+
+			return <>{ getChildren() }</>;
+		};
+
+		const { container, rerender } = render(
+			<Container>
+				{ ( {
+					onChange,
+				}: {
+					onChange: DateTimePickerControlOnChangeHandler;
+				} ) => {
+					return (
+						<DateTimePickerControl
+							dateTimeFormat={ dateTimeFormat }
+							currentDate={ originalDateTime.toISOString() }
+							onChange={ onChange }
+							onChangeDebounceWait={ 500 }
+						/>
+					);
+				} }
+			</Container>
+		);
+
+		const input = container.querySelector( 'input' );
+		userEvent.type(
+			input!,
+			'{selectall}{backspace}' + newDateTimeInputString
+		);
+
+		// re-render the component; we do this to then test whether our onChange still gets called
+		rerender(
+			<Container>
+				{ ( {
+					onChange,
+				}: {
+					onChange: DateTimePickerControlOnChangeHandler;
+				} ) => {
+					return (
+						<DateTimePickerControl
+							dateTimeFormat={ dateTimeFormat }
+							currentDate={ originalDateTime.toISOString() }
+							onChange={ onChange }
+							onChangeDebounceWait={ 500 }
+						/>
+					);
+				} }
+			</Container>
+		);
+
+		await waitFor(
```

Comment:
I prefer to use `useFakeTimers` from jest for performance reasons than waitFor.

Reply:
I'm all for improving the test performance. Since the tests are functioning now, and I'd ideally change out all the tests to use fake timers, what do you think about me making that change in a follow-up PR? That way, the only change in that PR will be the test changes, which will make verifying the PR (and the tests) just not have any regressions.

Reply:
Fair!